### PR TITLE
handle user budget period formatting error

### PR DESCRIPTION
**Pull request by elohmeier**
If the user enters a wrong period for a budget, e.g.
`2018-01-01 custom "budget" Expenses:Food "M" 300 EUR`
instead of 
`2018-01-01 custom "budget" Expenses:Food "monthly" 300 EUR`
fava returns a HTTP 500 and shows a NotImplementedError on the console, originating in fava/util/date.py", line 220, in number_of_days_in_period.

This PR adds a check in the budget parser to check for a valid period name and creates a Error object which assists the user with correcting the issue.

### CODE REVIEW

**Comment by yagebu** [comment]
Hey, thanks for the PR! Could you add a test for this?

File: fava/core/budgets.py [hunk 1]
```diff
@@ -61,6 +61,19 @@ def parse_budgets(custom_entries):
                                 entry.values[1].value,
                                 entry.values[2].value.number,
                                 entry.values[2].value.currency)
+                if budget.period not in [
+                    "daily", "weekly", "monthly", "quarterly", "yearly"
+                ]:
+                    errors.append(
+                        BudgetError(
+                            entry.meta,
+                            'invalid budget period "%s", please use "daily", \
```
Comment (aumayr) [inline fava/core/budgets.py]: Can you make the message translatable?

File: fava/core/budgets.py [hunk 2]
```diff
@@ -61,6 +61,19 @@ def parse_budgets(custom_entries):
...
+                        "weekly", "monthly", "quarterly" or "yearly"'
```
Comment (aumayr) [inline fava/core/budgets.py]: Should we put the possible values in the error message?

File: fava/core/budgets.py (hunk 1 above)
Comment (yagebu) [reply fava/core/budgets.py]: @aumayr: none of the other error messages are translateable - I wouldn't change that here

Comment (aumayr) [reply fava/core/budgets.py]: I think we should start to translate the error messages, at least the ones from Fava.

File: fava/core/budgets.py (hunk 2 above)
Comment (elohmeier) [reply fava/core/budgets.py]: I think it's the most user friendly option, while being clearly redundant.

**Comment by yagebu** [comment]
I've refactored the budget code to also use fava.util.date.Interval and added a similar check in #746. Thanks again

**Pull request closed without merging**